core/src/main/java/com/datastax/oss/driver/api/core/session/SessionBuilder.java

```diff
@@ -200,6 +199,17 @@ protected DriverContext buildContext(
     return new DefaultDriverContext(configLoader, typeCodecs);
   }
 
+  /**
+   * This <b>must</b> return an instance of {@code DefaultSession} (it's not expressed directly in
+   * the signature to avoid leaking that type through the protected API).
+   */
+  protected Session buildDefaultSession(
```

Comment:
I saw this warning in the comments of `buildContext` and had to do the same for this method. I think it is a code smell that indicates that we should rather extract an interface from this class and keep the implementation details in an internal class.